limitador/src/storage/infinispan/counters.rs

```diff
@@ -17,16 +17,49 @@
 use crate::storage::infinispan::response::response_to_string;
 use infinispan::errors::InfinispanError;
 use infinispan::{request, Infinispan};
+use std::convert::TryFrom;
 use std::time::Duration;
+use thiserror::Error;
+
+#[derive(Copy, Clone)]
+pub enum Consistency {
+    Weak,
+    Strong,
+}
+
+#[derive(Error, Debug)]
+#[error("invalid consistency mode: {mode}")]
+pub struct InvalidConsistencyErr {
+    mode: String,
+}
+
+impl TryFrom<String> for Consistency {
+    type Error = InvalidConsistencyErr;
+
+    fn try_from(mut value: String) -> Result<Self, Self::Error> {
+        value.make_ascii_lowercase();
+
+        match value.as_str() {
+            "weak" => Ok(Consistency::Weak),
+            "strong" => Ok(Consistency::Strong),
+            val => Err(InvalidConsistencyErr { mode: val.into() }),
```

Comment:
You might want to use `_ => Err(InvalidConsistencyErr { mode: value }),` to avoid an extra clone.

And _perhaps_ you can add `#[from]` to `mode: String,` above and do here `_ => Err(value.into()),`, though I'm not sure.

Reply:
Done :+1: